src/auspex/plotting.py

```diff
@@ -6,66 +6,78 @@
 #
 #    http://www.apache.org/licenses/LICENSE-2.0
 
-import threading
+from threading import Thread
 import subprocess
 import psutil
 import os
+import json
 import sys
 import tempfile
 import time
-
+import asyncio
+import zmq
+import zmq.asyncio
 from auspex.log import logger
 
-class BokehServerProcess(object):
-    def __init__(self, notebook=False):
-        super(BokehServerProcess, self).__init__()
-        self.run_in_notebook = notebook
-        self.pid_filename = os.path.join(tempfile.gettempdir(), "auspex_bokeh.pid")
+class MatplotServerThread(Thread):
 
-    def run(self):
-        # start a Bokeh server if one is not already running
-        pid = self.read_session_pid()
-        if pid:
-            self.p = psutil.Process(pid)
-            logger.info("Using existing Bokeh server")
-            return
-        logger.info("Starting Bokeh server")
-        args = ["bokeh", "serve", "--port", "5006", "--allow-websocket-origin=localhost:8888", "--allow-websocket-origin=localhost:8889", "--allow-websocket-origin=localhost:8890"]
-        self.p = subprocess.Popen(args, env=os.environ.copy())
-        self.write_session_pid()
-        # sleep to give the Bokeh server a chance to start
-        # TODO replace this with some bokeh client API call that
-        # verifies that the server is running
-        time.sleep(3)
+    def __init__(self, plot_desc={}, status_port = 7771, data_port = 7772):
+        super(MatplotServerThread, self).__init__()
+        self.plot_desc = plot_desc
+        self.status_port = status_port
+        self.data_port = data_port
+        self.daemon = True
+        self.stopped = False
+        self.start()
+
+    async def poll_sockets(self):
```

Comment:
I'd love to use `self._loop.add_reader(...)` instead of this polling method, but there are some ominous looking comment on http://pyzmq.readthedocs.io/en/latest/eventloop.html about the file-descriptor interface:
> It is possible to integrate zmq sockets into existing eventloops without modifying the poller by using the socket.FD attribute. The incredibly unfortunate aspect of this is that it was implemented as an edge-triggered fd, which is highly error prone, and I wouldn’t recommend using unless absolutely necessary. This is used in zmq.green, and has been the source of many problems.

There is a `ZMQStream.on_recv()` which doesn't carry that warning, but I can't tell if it is Tornado-only.